src/ripple/net/impl/FileDownloader.cpp

```diff
@@ -0,0 +1,82 @@
+#include <ripple/net/FileDownloader.h>
+
+namespace ripple
+{
+
+FileDownloader::FileDownloader(
+    boost::asio::io_service & io_service,
+    beast::Journal j,
+    Config const & config)
+    : SSLHTTPDownloader(io_service, j, config)
+{
+}
+
+auto
+FileDownloader::getParser(boost::filesystem::path dstPath,
+    std::function<void(boost::filesystem::path)> complete,
+    boost::system::error_code & ec) -> std::shared_ptr<parser>
+{
+    using namespace boost::beast;
+
+    auto p = std::make_shared < http::response_parser < http::file_body >> ();
+    p->body_limit(std::numeric_limits<std::uint64_t>::max());
+    p->get().body().open(
+        dstPath.string().c_str(),
+        boost::beast::file_mode::write,
+        ec);
+    if(ec)
+    {
+        p->get().body().close();
+        fail(dstPath, complete, ec, "open");
+    }
+
+    return p;
+}
+
+bool
+FileDownloader::checkPath(boost::filesystem::path const & dstPath)
+{
+    try
+    {
+        if(exists(dstPath))
+        {
+            JLOG(j_.error()) <<
+                             "Destination file exists";
+            return false;
+        }
+    }
+    catch(std::exception const & e)
+    {
+        JLOG(j_.error()) <<
+                         "exception: " << e.what();
```

Comment:
Consider logging where the exception was caught.
eg.
```
JLOG(j_.error()) <<
    "exception " << e.what() << " in function " << __func__;
```

Reply:
Done